### ISSUE TITLE
Optimize plugin architecture

### ISSUE THREAD

**Issue by hirasso** [swup/swup#708]
**Describe the problem**

A few ideas popped up for improvement the current architecture of @swup/plugin. This became apparent when developing the first TypeScript plugin ([fragment plugin](https://github.com/swup/fragment-plugin)).

**Ideas** 

- Include plugin base class in core with named export
- Don't require swup in the plugin base class
- Pass `swup` instance into the plugin class somehow

**Comment by daun** [comment]
@hirasso Is this still a thing? I feel like the plugin system has worked out quite well, especially after porting them all over to TypeScript. I don't see much need for improving on it at this moment, but would love to know how you're feeling about it.

**Comment by hirasso** [comment]
Good idea. Let's close this - we can always reopen or create a new issue should we need any improvements in the future.
